tests/unit/cli/registry/test_cli_registry.py

```diff
@@ -0,0 +1,98 @@
+        mock_parse_model_template.return_value = mock_templates[0]
+
+        with pytest.raises(RuntimeError):
+            r = Registry()
+
+        r = Registry(templates_dir=".")
+        assert r.templates == [mock_templates[0]]
+
+        r = Registry(templates_dir=".", experimental=True)
+        assert r.templates == [mock_templates[0]] * 2
+
+        r = Registry(templates=mock_templates)
+        assert r.templates == mock_templates
+
+    @e2e_pytest_unit
+    def test_filter(self, mock_templates):
+        """Check TestRegistry.filter function working well.
+
+        <Steps>
+            1. Create an instance of Registry with the mock templates
+            2. Test filtering by framework
+            3. Test filtering by task_type
+            4. Test filtering by both framework and task_type
+        """
+        registry = Registry(templates=mock_templates)
+
+        filtered_registry = registry.filter(framework="OTX001")
+        assert len(filtered_registry.templates) == 2
+        assert filtered_registry.templates[0].task_type == "Classification"
+        assert filtered_registry.templates[1].task_type == "Detection"
+        filtered_registry = registry.filter(framework="OTX002")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Segmentation"
+
+        filtered_registry = registry.filter(task_type="Detection")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Detection"
+
+        filtered_registry = registry.filter(framework="OTX002", task_type="Segmentation")
+        assert len(filtered_registry.templates) == 1
+        assert filtered_registry.templates[0].task_type == "Segmentation"
+
+    @e2e_pytest_unit
+    def test_get(self, mock_templates):
+        registry = Registry(templates=mock_templates)
+        template = registry.get("003")
+        assert template.task_type == "Detection"
+
+        with pytest.raises(ValueError):
+            registry.get("004")
+
+    @e2e_pytest_unit
+    def test_get_backbones(self, mocker, mock_templates):
+        mock_get_backbone_list = mocker.patch("otx.cli.registry.registry.get_backbone_list")
+        mock_get_backbone_list.return_value = ["resnet50", "resnet101"]
+        r = Registry(templates=mock_templates)
+        assert r.get_backbones(["torch", "tensorflow"]) == {
+            "torch": ["resnet50", "resnet101"],
+            "tensorflow": ["resnet50", "resnet101"],
+        }
+
+
+@e2e_pytest_unit
+def test_find_and_parse_model_template(mocker):
+    mock_template = mocker.MagicMock(framework="test", task_type="test", model_template_id="001")
+    mock_parse_model_template = mocker.patch("otx.cli.registry.registry.parse_model_template")
+    mock_parse_model_template.return_value = mock_template
+
+    assert find_and_parse_model_template("001") == mock_template
+
+    mock_exists = mocker.patch("os.path.exists")
```

Comment:
I think it's better to separate this part as other test function because this is different with working well case. How do you think about?

Reply:
I think it's ok to keep this one as it's a function with the same purpose. What do you think?

Reply:
I think this test includes two cases. One is `os.path.exists(path_or_id)` is `True` and other is False. So, I think it's better to separate them.

Reply:
This has been reflected.